@gantoine yeah I do see some failures -

`'You must set OIDC_RSA_PRIVATE_KEY to use RSA algorithm'`

and

```
FAILED posthog/hogql/database/schema/test/test_persons_revenue_analytics.py::TestPersonsRevenueAnalytics::test_query_revenue_analytics_table_sources_with_managed_viewsets_ff - AssertionError: Lists differ ...
First differing element 0:
(UUID('00000000-0000-4000-8000-000000000002'), Decimal('4161.34422'), Decimal('0'))
(UUID('00000000-0000-4000-8000-000000000001'), Decimal('482.2158673452'), Decimal('16.3052916666'))
```

I don't see how they could be related to this library. The first error is from Django OAuth/OIDC and second looks like some data issue in revenue analytics. The paramiko change only affects SSH tunnel connections for data warehouse imports.

--- a/.github/workflows/ci-hobby.yml
+++ b/.github/workflows/ci-hobby.yml
@@ -254,7 +254,7 @@ jobs:
                           body: errorBody,
                       });
             - name: Fail if Docker build failed
-              if: steps.wait-for-image.outputs.conclusion != 'success'
+              if: steps.wait-for-image.outputs.conclusion != 'success' && steps.wait-for-image.outputs.conclusion != 'skipped'
               run: |
                   echo "Docker image build failed with conclusion: ${{ steps.wait-for-image.outputs.conclusion }}"
                   exit 1

--- a/posthog/api/test/test_toolbar_oauth_authorize.py
+++ b/posthog/api/test/test_toolbar_oauth_authorize.py
@@ -2,7 +2,13 @@ from urllib.parse import parse_qs, urlparse
 
 from posthog.test.base import APIBaseTest
 
+from django.conf import settings
+from django.test import override_settings
 
+from posthog.api.oauth.test_dcr import generate_rsa_key
+
+
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestToolbarOAuthAuthorize(APIBaseTest):
     def setUp(self):
         super().setUp()

--- a/posthog/api/test/test_toolbar_oauth_endpoint_auth.py
+++ b/posthog/api/test/test_toolbar_oauth_endpoint_auth.py
@@ -13,10 +13,12 @@ from datetime import timedelta
 from posthog.test.base import APIBaseTest
 
 from django.conf import settings
+from django.test import override_settings
 from django.utils import timezone
 
 from parameterized import parameterized
 
+from posthog.api.oauth.test_dcr import generate_rsa_key
 from posthog.models.oauth import OAuthAccessToken, OAuthApplication
 
 
@@ -43,6 +45,7 @@ def _make_token(user, app, token_str, scope="*", delta_hours=1):
     )
 
 
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestToolbarEndpointOAuthAuth(APIBaseTest):
     """
     Every toolbar-consumed endpoint should accept OAuth access tokens
@@ -195,6 +198,7 @@ class TestToolbarEndpointOAuthAuth(APIBaseTest):
         )
 
 
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestUploadedMediaOAuthAuth(APIBaseTest):
     """uploaded_media is tested separately — its only toolbar action is POST (upload)."""
 
@@ -248,6 +252,7 @@ class TestUploadedMediaOAuthAuth(APIBaseTest):
         assert response.status_code != 401
 
 
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestHedgehogConfigOAuthAuth(APIBaseTest):
     """hedgehog_config uses /api/users/@me/ path, not team-scoped, so tested separately."""
 
@@ -298,6 +303,7 @@ class TestHedgehogConfigOAuthAuth(APIBaseTest):
         assert response.status_code in (401, 403)
 
 
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestToolbarOAuthScopesConfig(APIBaseTest):
     """Verify TOOLBAR_OAUTH_SCOPES covers every toolbar endpoint scope."""
 

--- a/posthog/api/test/test_toolbar_oauth_primitives.py
+++ b/posthog/api/test/test_toolbar_oauth_primitives.py
@@ -13,6 +13,7 @@ from rest_framework import status
 from rest_framework.exceptions import AuthenticationFailed
 from rest_framework.request import Request
 
+from posthog.api.oauth.test_dcr import generate_rsa_key
 from posthog.api.oauth.toolbar_service import (
     CALLBACK_PATH,
     ToolbarOAuthError,
@@ -24,6 +25,7 @@ from posthog.auth import TemporaryTokenAuthentication
 from posthog.models import Organization, Team, User
 
 
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestToolbarOAuthPrimitives(APIBaseTest):
     def setUp(self):
         super().setUp()
@@ -694,6 +696,7 @@ class TestTokenEndpointStatusRemapping(APIBaseTest):
         assert cm.exception.status_code == expected_status
 
 
+@override_settings(OAUTH2_PROVIDER={**settings.OAUTH2_PROVIDER, "OIDC_RSA_PRIVATE_KEY": generate_rsa_key()})
 class TestToolbarOAuthCallbackExchange(APIBaseTest):
     def setUp(self):
         super().setUp()

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -94,7 +94,7 @@ dependencies = [
     "openpyxl==3.1.2",
     "orjson==3.11.5",
     "pandas~=2.2.2",
-    "paramiko==3.4.0",
+    "paramiko==3.4.1",
     "phonenumberslite==8.13.6",
     "pillow==12.1.1",
     "posthoganalytics==7.9.0",

